src/internal/sdata/sdata_test.go

```diff
@@ -20,6 +20,86 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/internal/testsnowflake"
 )
 
+// TestFormatParse is a round trip from a Tuple through formatting and parsing
+// back to a Tuple again.
+func TestFormatParse(t *testing.T) {
+	testCases := []struct {
+		Name string
+		NewW func(w io.Writer, fieldNames []string) TupleWriter
+		NewR func(r io.Reader, fieldNames []string) TupleReader
+	}{
+		{
+			Name: "CSV",
+			NewW: func(w io.Writer, _ []string) TupleWriter {
+				return NewCSVWriter(w, nil)
+			},
+			NewR: func(r io.Reader, _ []string) TupleReader {
+				return NewCSVParser(r)
+			},
+		},
+		{
+			Name: "JSON",
+			NewW: func(w io.Writer, fieldNames []string) TupleWriter {
+				return NewJSONWriter(w, fieldNames)
+			},
+			NewR: func(r io.Reader, fieldNames []string) TupleReader {
+				return NewJSONParser(r, fieldNames)
+			},
+		},
+	}
+	newTuple := func() Tuple {
+		a := int64(0)
+		b := float64(0)
+		c := ""
+		d := sql.NullInt64{}
+		return Tuple{&a, &b, &c, &d}
+	}
+	fieldNames := []string{"a", "b", "c", "d"}
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			const N = 10
+			buf := &bytes.Buffer{}
+			fz := fuzz.New()
```

Comment:
Do we need to create new fuzzer for each test case? Can we just create a single fuzzer outside the loop?

Reply:
I was doing this so it would be deterministic and produce the same values for each test.  We also need to set the rand.Source, which I added.

Reply:
Just for my own understanding, but I'd assume that you only need to set the random seed once for all fuzzing operations to be deterministic?

Reply:
If we want to get the same data for each run of the test we have to seed it with the same source every run.